### PR TITLE
feat: add a workflow to check for unused dependencies

### PR DESCRIPTION
**Pull request by huitseeker**
This should help close e.g. https://github.com/lurk-lab/lurk-rs/issues/983 downstream.

### CODE REVIEW

File: .github/templates/UNUSED_DEPS_ISSSUE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: "chore: some installed deps are not needed"
+labels: automated-issue
+---
+
+Some dependencies specified in `Cargo.toml` are not needed.
+
+Check the [unused dependencies sanity check]({{env.WORKFLOW_URL}}) workflow for details.
+
+This issue was raised by the workflow at `.github/workflows/sanity.yml`.
```
Comment (samuelburnham) [inline .github/templates/UNUSED_DEPS_ISSSUE.md:10]: I think this should be `unused-deps.yml`

File: .github/workflows/unused-deps.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  unused-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      # udeps requires nightly
+      - uses: dtolnay/rust-toolchain@nightly
+      - uses: taiki-e/install-action@cargo-udeps
+      - name: Check for unused dependencies
+        run: cargo udeps --lib --features "${{ inputs.repository }}"
+      - uses: JasonEtco/create-an-issue@v2
```
Comment (samuelburnham) [inline .github/workflows/unused-deps.yml:30]: Should we standardize usage of this action over https://github.com/peter-evans/create-or-update-comment? Both seem workable but I like this one for the requirement of a formatted issue template

File: .github/workflows/unused-deps.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Runs a series of sanity checks for crate consumers.
+
+name: sanity
```
Comment (samuelburnham) [inline .github/workflows/unused-deps.yml:3]: Nit: If we are only performing a `udep` check then the name could be more descriptive, e.g. `Unused dependency check`, as this name is load-bearing in the Actions overview. If there are plans to add more/consolidate with other checks then e.g. `Sanity checks` works.

**Review by samuelburnham — CHANGES_REQUESTED** [review]
Looks great, only blocking on link fix

File: .github/workflows/unused-deps.yml (hunk 2 above)
Comment (huitseeker) [reply .github/workflows/unused-deps.yml:30]: I upgraded to this for precisely that reason. I'm up for upgrading to JasonEtco/create-an-issue across the board, but I haven't noticed any instances of peter-evans/create-or-update-comment in the present repo.

**Review by samuelburnham — DISMISSED** [review]

File: .github/templates/UNUSED_DEPS_ISSSUE.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+---
+title: "chore: some installed deps are not needed"
+labels: automated-issue
+---
+
+Some dependencies specified in `Cargo.toml` are not needed.
+
+Check the [unused dependencies sanity check]({{env.WORKFLOW_URL}}) workflow for details.
+
+This issue was raised by the workflow at `.github/workflows/unused-deps.yml`.
```
Comment (samuelburnham) [inline .github/templates/UNUSED_DEPS_ISSSUE.md:10]: Nit: Should we include the full URL like in #5, since the issue will be opened in another repo?

**Review by samuelburnham — APPROVED** [review]

**Pull request merged**